app/javascript/topNavigation/utilities.js

```diff
@@ -145,11 +145,15 @@ export async function getInstantClick(waitTime = 2000) {
  */
 export function initializeMobileMenu(menuTriggers, moreMenus) {
   menuTriggers.forEach((trigger) => {
-    trigger.addEventListener('click', toggleBurgerMenu);
+    if (trigger.onclick === null) {
+      trigger.onclick = toggleBurgerMenu;
+    }
```

Comment:
So this resolves the issue you caught @aitchiss, which happened because InstantClick would add a second click event on top of the original. So the toggle would go `open -> close` instantly after the second page navigation.

I'm not in love with this solution but it works... I guess an alternative would be to add a ` data-listener-attached="false"` to the element and then check for that in the listener function:
```js
function toggleBurgerMenu(e) {
  if(e.dataset.listenerAttached === "false") {
    // rest of the code
  }
}
```

Reply:
if you use `onclick = ` instead of `addEventListener` you can safely omit the `null` check, since there can only be one `onclick` property at a time (so if InstantClick tries to attach the same handler again, it just safely overwrites the first one instead of adding a second 🙂)

https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onclick#value